api/client/gro_client.py

```diff
@@ -82,6 +88,54 @@ def get_df(self):
                 self._data_frame = self._data_frame.merge(tmp, how='outer')
         return self._data_frame
 
+    def GDH(self, gdh_selection):
+        """Wrapper for get_data_points with alternative input and output style.
+
+        The selection of data series to retrieve is encoded in a
+        'gdh_seletion' string of the form
+        <metric_id>-<item_id>-<region_id>-<partner_region_id>-<source_id>-<frequency_id>
+
+        For example, client.GDH("860032-274-1231-0-14-9") will get the
+        data points for Production of Corn in China from PS&D at an
+        annual frequency, e.g.
+        for csv_row in client.GDH("860032-274-1231-0-14-9"):
+            print csv_row
+
+        Parameters:
+        ----------
+        gdh_selection: string
+
+        Returns:
+        ------
+        DataFrame
+
+            subset of the main DataFrame get_df() with only the requested series.
+
+        """
+        entity_keys = ['metric_id', 'item_id', 'region_id', 'partner_region_id',
+                       'source_id', 'frequency_id']
+        entity_ids = [int(x) for x in gdh_selection.split('-')]
+        selection = dict(zip(entity_keys, entity_ids))
+        self.add_single_data_series(selection)
+        df = self.get_df()
+        # Subset the dataframe and fill in names
+        pairs = self.get_names_for_selection(selection)
+        unit_id = df['unit_id'].values[0]
+        unit_name = self.lookup("units", unit_id).get('name')
+        pairs.append(("unit", unit_name))
+        cols = pandas.MultiIndex.from_tuples([tuple(pair[1] for pair in pairs)],
+                                             names=tuple(pair[0] for pair in pairs))                                                          
+        filters = ((df['item_id'] == selection['item_id']) & 
+                   (df['metric_id'] == selection['metric_id']) &
+                   (df['region_id'] == selection['region_id']) &
+                   (df['frequency_id'] == selection['frequency_id']))
+        if 'partner_region_id' in df.columns:
+            filters = (filters & (df['partner_region_id'] == selection['partner_region_id']))
+        series = pandas.DataFrame(
```

Comment:
@nemozen  What's the df index issue you mentioned earlier? 

but right now output will break if data is from expansion , this filter will exclude all rows unless it's at_level, unless we aggregate output here.  not sure what's the desired behavior here.